notifier/selfstate/checkable/checkabletypes.go

```diff
@@ -0,0 +1,84 @@
+package checkable
+
+import (
+	"fmt"
+	"github.com/moira-alert/moira"
+)
+
+const (
+	RedisDisconnectedErrorMessage  = "Redis disconnected"
+	FilterStateErrorMessage        = "Moira-Filter does not receive metrics"
+	CheckerStateErrorMessage       = "Moira-Checker does not check triggers"
+	RemoteCheckerStateErrorMessage = "Moira-Remote-Checker does not check remote triggers"
+)
+
+const templateMoreThan = "%s more than %ds. Send message."
+
+// Checkable is the interface for simplified events verification.
+type Checkable interface {
+	Check(int64) *moira.NotificationEvent
+	NeedTurnOffNotifier() bool
+	NeedToCheckOthers() bool
+}
+
+// baseDelay basic structure for Checkable structures
+type baseDelay struct {
+	log moira.Logger
+	db  moira.Database
+
+	delay       int64
+	last, count *int64
+}
+
+// Init data in base structure
+func (base *baseDelay) Init(b baseDelay, last, count *int64, delay int64) {
+	*base = b
+	base.last, base.count, base.delay = last, count, delay
+}
+
+// GetBaseDelay returning the base structure to creating structures Checkable
+func GetBaseDelay(log moira.Logger, db moira.Database) *baseDelay {
+	return &baseDelay{
+		log: log,
+		db:  db,
+	}
+}
+
+// RedisDelay structure to check redis disconnect delay
```

Comment:
Lets use sentences in documentation. 
Reference: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

For example I think this will be much better: `// RedisDelay is a structure to check redis disconnect delay`